### ISSUE TITLE
Precompilation fails on Julia v1.7.0-rc1

### ISSUE THREAD

**Issue by sethaxen** [JuliaDiff/Diffractor.jl#61]
```julia
julia> import Pkg; Pkg.precompile()
Precompiling project...
  ✗ Diffractor
  0 dependencies successfully precompiled in 2 seconds (24 already precompiled)

ERROR: The following 1 direct dependency failed to precompile:

Diffractor [9f5e2b26-1114-432f-b630-d3fe2085c51c]

Failed to precompile Diffractor [9f5e2b26-1114-432f-b630-d3fe2085c51c] to /Users/sethaxen/.julia/compiled/v1.7/Diffractor/jl_Nt5Rnu.
ERROR: LoadError: UndefVarError: @constprop not defined
Stacktrace:
 [1] include(mod::Module, _path::String)
   @ Base ./Base.jl:420
 [2] include(x::String)
   @ Diffractor ~/.julia/packages/Diffractor/HYuxt/src/Diffractor.jl:1
 [3] top-level scope
   @ ~/.julia/packages/Diffractor/HYuxt/src/Diffractor.jl:7
 [4] include
   @ ./Base.jl:420 [inlined]
 [5] include_package_for_output(pkg::Base.PkgId, input::String, depot_path::Vector{String}, dl_load_path::Vector{String}, load_path::Vector{String}, concrete_deps::Vector{Pair{Base.PkgId, UInt64}}, source::Nothing)
   @ Base ./loading.jl:1318
 [6] top-level scope
   @ none:1
 [7] eval
   @ ./boot.jl:373 [inlined]
 [8] eval(x::Expr)
   @ Base.MainInclude ./client.jl:453
 [9] top-level scope
   @ none:1
```

**Comment by simeonschaub** [comment]
Yeah, you will need the latest nightly or build directly off the `backports-release-1.7` branch since the name of `@constprop` wae just recently changed.

**Comment by Moelf** [comment]
it compiles on released 1.7
